src/Lecture1.hs

```diff
@@ -94,7 +101,16 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> String -> String
+subString start end str = result start end str
+    where 
+        result :: Int -> Int -> String -> String
+        result start end str = 
+            if end < 0 then 
+                ""
+            else if start < 0 then 
+                drop 0 (take (end + 1) str)
```

Comment:
I believe you wrote this for symmetricity reasons with the last case 🙂 
Since `drop 0 list` is the same as `list`, you can remove `drop 0` from here. But it's probably won't result in any noticeable performance impact.

```suggestion
                take (end + 1) str
```